examples/chip-tool-darwin/commands/common/CHIPCommandStorageDelegate.mm

```diff
@@ -11,10 +11,10 @@ id CHIPGetDomainValueForKey(NSString * domain, NSString * key)
     return nil;
 }
 
-void CHIPSetDomainValueForKey(NSString * domain, NSString * key, id value)
+BOOL CHIPSetDomainValueForKey(NSString * domain, NSString * key, id value)
 {
     CFPreferencesSetAppValue((CFStringRef) key, (__bridge CFPropertyListRef _Nullable)(value), (CFStringRef) domain);
-    CFPreferencesAppSynchronize((CFStringRef) domain);
+    return CFPreferencesAppSynchronize((CFStringRef) domain) == true;
```

Comment:
is the `== true` doing anything useful here?

Reply:
It is!  `CFPreferencesAppSynchronize` returns `Boolean`, which is a signed type.  It's documented as:

> true if synchronization was successful, otherwise false. 

This function returns `BOOL`, which is an unsigned type.  Just returning a `Boolean` from it errors out from `-Wconversion` warnings.  Hence this silliness.